pennylane/labs/lie/structure_constants.py

```diff
@@ -0,0 +1,80 @@
+"""A function to compute the structure constants that make up the adjoint representation of a Lie algebra"""
+
+import numpy as np
+
+from pennylane.typing import TensorLike
+
+
+def structure_constants_dense(g: TensorLike, is_orthogonal: bool = False) -> TensorLike:
+    r"""
+    Compute the structure constants that make up the adjoint representation of a Lie algebra.
+
+    This function computes the structure constants of a Lie algebra provided by their dense matrix representation,
+    obtained from, e.g., :func:`~lie_closure_dense`.
+    This is sometimes more efficient than using the sparse Pauli representations of :class:`~PauliWord` and
+    `~PauliSentence` that are employed in :func:`~structure_constants`, e.g., when there are few generators
+    that are sums of many Paulis.
+
+    .. seealso:: For details on the mathematical definitions, see :func:`~structure_constants` and the section "Lie algebra basics" in our `g-sim demo <https://pennylane.ai/qml/demos/tutorial_liesim/#lie-algebra-basics>`__.
+
+    Args:
+        g (np.array): The (dynamical) Lie algebra provided as dense matrices, as generated from :func:`pennylane.labs.lie.lie_closure_dense`.
+            ``g`` should have shape ``(d, 2**n, 2**n)`` where ``d`` is the dimension of the algebra and ``n`` is the number of qubits.
+        is_orthogonal (bool): Whether or not the matrices in ``g`` are orthogonal with respect to the Hilbert-Schmidt inner product on
+            (skew-)Hermitian matrices. If the inputs are orthogonal, it is recommended to set ``is_orthogonal`` to ``True`` to reduce
+            computational cost. Defaults to ``False``.
+
+    Returns:
+        TensorLike: The adjoint representation of shape ``(d, d, d)``, corresponding to indices ``(gamma, alpha, beta)``.
+
+    **Example**
+
+    Let us generate the DLA of the transverse field Ising model using :func:`~lie_closure_dense`.
+
+    >>> n = 4
+    >>> gens = [X(i) @ X(i+1) + Y(i) @ Y(i+1) + Z(i) @ Z(i+1) for i in range(n-1)]
+    >>> g = lie_closure_dense(gens)
+    >>> g.shape
+    (12, 16, 16)
+
+    The DLA is represented by a collection of twelve :math:`2^4 \times 2^4` matrices.
+    Hence, the dimension of the DLA is :math:`d = 12` and the structure constants have shape ``(12, 12, 12)``.
+
+    >>> adj = structure_constants_dense(g)
+    >>> adj.shape
+    (12, 12, 12)
+
+    """
+    dimg, chi, _ = g.shape
+    assert g.shape[2] == g.shape[1]
+
+    # compute all commutators by computing all products first according to "aij,bjk->abik"
+    prod = np.moveaxis(np.tensordot(g, g, axes=[[2], [1]]), 1, 2)
+    all_coms = np.reshape(prod - prod.transpose((1, 0, 2, 3)), (-1, chi, chi))
+
+    # project commutators on basis of g
+    # vectorized computation to obtain coefficients in decomposition:
+    # v = ∑ (tr(v @ e_j) / ||e_j||^2) * e_j
+    if is_orthogonal:
+        # Compute norms of orthogonal entries
+        norms = np.einsum("aij,aji->a", g, g).real
+        gram_inv = np.diag(1 / norms)
```

Comment:
why do we need to compute any norms if the inputs are orthogonal, at all?

Reply:
That one here is just the identity matrix in the case the basis is orthonormal 👍

Reply:
ah I just see that we made it `is_orthogonal`. Wouldnt `is_orthonormal` make more sense, since it is more about normalization here and from `lie_closure_dense` we are getting orthonormal bases

Reply:
Ah true! I was confused about the generic Pauli Word scenario that leads to orthogonal inputs and the actual output of `lie_closure_dense`.
I'll edit it.